src/dali_executor/dali_pipeline.h

```diff
@@ -76,7 +76,8 @@ class DaliPipeline {
       serialized_pipeline_(serialized_pipeline),
       max_batch_size_(max_batch_size),
       num_threads_(num_threads),
-      device_id_(device_id) {
+      device_id_(device_id),
+      no_gpu_(device_id_ < 0) {
```

Comment:
Maybe a `NoGpu()`/`no_gpu()` method instead of a separate field?

Reply:
Done